portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+
+/*
+ * Toggles the drop down text.
+ */
+function toggleDropDown(id) {
+  document.getElementById(id).classList.toggle("show");
+}
+
+/*
+ * Adds a random favorite quote to the page.
+ */
+function addRandomFavoriteQuote() {
+  const favoriteQuotes = [
+    'Many years later, as he faced the firing squad, Colonel Aureliano Buendía was to remember that distant afternoon when his father took him to discover ice. \n —Gabriel García Márquez, One Hundred Years of Solitude',
+    'Hope is a dangerous thing. Hope can drive a man insane. \n - The Shawshank Redemption',
+    'Bran thought about it. "Can a man still be brave if he\'s afraid?" "That is the only time a man can be brave," his father told him. \n ― George R.R. Martin, A Game of Thrones',
+  ];
+
+  // Picks a random quote.
+  const randomFavoriteQuote = favoriteQuotes[Math.floor(Math.random() * favoriteQuotes.length)];
+
+  // Adds the favorite quote to the page.
+  const favoriteQuoteContainer = document.getElementById('favorite-quote-container');
+  favoriteQuoteContainer.innerText = randomFavoriteQuote;
+}
+
+/*
+ * Adds a random fun fact to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomFunFact() {
+  const funFacts = [
+    'sister', 
+    'selfie',
+    'I\'m a fan of collections. I collect postcards, stickers, books, snacks, etc.',
+    'My English name is Heidi. Many people can\'t pronounce that.',
+    'Funny enough that most of my Chinese friends call me by my English name while my American friends call me by my Chinese name.',
+  ];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  // Picks a random index for the fun fact.
+  const index = Math.floor(Math.random() * funFacts.length);
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  // Adds the fun fact to the page.
+  const funFactContainer = document.getElementById('fun-fact-container');
+  if (index == 0 || index == 1) {
+    funFactContainer.innerHTML = document.getElementById('fun-fact-' + funFacts[index]).innerText;
+  } else {
+    funFactContainer.innerText = funFacts[index];
+  }
```

Comment:
I think you have two different types of fun facts:
1) js scripts
2) regular text

For the purpose of code health, I'd refactor the code in this:

const funFactsJsIds = [
    'fun-fact-sister',
    'fun-fact-selfie',
];

const funFactsStrings = [
    'I\'m a fan of collections. I collect postcards, stickers, books, snacks, etc.',
    'blahblah',
]

const index = Math.floor(Math.random() * (funFactsJsIds.length + funFactsStrings.length));

if (index < funFactsJsIds.length) {
   funFactContainer.innerHTML = document.getElementById(funFactsJsIds[index]).innerText;
} else {
    funFactContainer.innerText = funFacts[index - funFactsStrings.length];
}


In this way, you don't have to keep the index == 0 || index == 1 which is very fragile when you add more js in the future.

Reply:
Got it. Thanks!